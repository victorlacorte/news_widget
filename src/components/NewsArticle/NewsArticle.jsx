import React from 'react';
import PropTypes from 'prop-types';

import date from 'commons/utils/date';

import {
  Root,
  LinkTitle,
  Container,
  PublishedAt,
  SourceName,
} from './styles';

function NewsArticle(props) {
  const {
    title,
    url,
    publishedAt,
    sourceName,
    ...otherProps
  } = props;

  return (
    <Root {...otherProps}>
      <LinkTitle
        href={url}
        target="_blank"
        rel="noopener noreferrer"
      >
        {title}
      </LinkTitle>

      <Container>
        <PublishedAt>{date.convertIsoToBrLocale(publishedAt)}</PublishedAt>
        <SourceName>{sourceName}</SourceName>
      </Container>
    </Root>
  );
}

NewsArticle.defaultProps = {
  sourceName: 'N/A',
};

NewsArticle.propTypes = {
  title: PropTypes.string.isRequired,

  url: PropTypes.string.isRequired,

  publishedAt: PropTypes.string.isRequired,

  sourceName: PropTypes.string,
};

export default NewsArticle;
