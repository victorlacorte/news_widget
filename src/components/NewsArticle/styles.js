import styled, { css } from 'styled-components';

const Root = styled.div`
  // overflow-wrap: break-word;
  border-bottom: 2px solid ${({ theme }) => theme.palette.grey.dark};
  padding-bottom: 24px;
  margin-bottom: 24px;
`;

const LinkTitle = styled.a`
  ${({ theme }) => css`
    ${theme.typography.h2}
  `}

  color: ${({ theme }) => theme.palette.black.main};
  text-decoration: none;
`;

const Container = styled.div`
  display: flex;
  margin-top: 24px;

  ${({ theme }) => theme.breakpoints.up('sm')`
    & > *:first-child {
      margin-right: 24px;
      align-items: center;
    }
  `}

  ${({ theme }) => theme.breakpoints.down('sm')`
    flex-direction: column;

    & > *:first-child {
      margin-bottom: 8px;
    }
  `}
`;

const PublishedAt = styled.div`
  ${({ theme }) => css`
    ${theme.typography.body}
  `}

  color: ${({ theme }) => theme.palette.grey.dark};
`;

const SourceName = styled.div`
  ${({ theme }) => css`
    ${theme.typography.body}
  `}

  color: white;
  background-color: ${({ theme }) => theme.palette.grey.main};
  padding: 4px 36px;
  border-radius: 4px;
  width: fit-content;
  text-align: center;
`;

export {
  Root,
  LinkTitle,
  Container,
  PublishedAt,
  SourceName,
};
