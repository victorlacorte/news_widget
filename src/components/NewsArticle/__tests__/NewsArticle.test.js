import React from 'react';
import { shallow } from 'enzyme';

import NewsArticle from '../NewsArticle';

describe('<NewsArticle />', () => {
  it('creates a NewsArticle component', () => {
    const props = {
      title: 'Foo',
      url: 'www.foo.bar',
      publishedAt: '2019-11-08T11:00:00Z',
      sourceName: 'Bar',
    };

    const wrapper = shallow(<NewsArticle {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
