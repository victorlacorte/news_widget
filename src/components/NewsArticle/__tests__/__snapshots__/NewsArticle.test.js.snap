// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsArticle /> creates a NewsArticle component 1`] = `
<styled.div>
  <styled.a
    href="www.foo.bar"
    rel="noopener noreferrer"
    target="_blank"
  >
    Foo
  </styled.a>
  <styled.div>
    <styled.div>
      08/11/2019
    </styled.div>
    <styled.div>
      Bar
    </styled.div>
  </styled.div>
</styled.div>
`;
